env/io_posix.cc

```diff
@@ -872,6 +872,75 @@ IOStatus PosixRandomAccessFile::InvalidateCache(size_t offset, size_t length) {
 #endif
 }
 
+IOStatus PosixRandomAccessFile::ReadAsync(
+    const IOOptions& /*opts*/, IODebugContext* /*dbg*/, FSReadRequest* req,
+    std::function<void(FSReadResponse* resp)> cb, IOHandle* io_handle) {
+  IOStatus io_s;
+  if (use_direct_io()) {
+    assert(IsSectorAligned(req->offset, GetRequiredBufferAlignment()));
+    assert(IsSectorAligned(req->len, GetRequiredBufferAlignment()));
+    assert(IsSectorAligned(req->scratch, GetRequiredBufferAlignment()));
+  }
+
+#if defined(ROCKSDB_IOURING_PRESENT)
+  // Step 1: io_uring_queue_init
+  struct io_uring* iu = nullptr;
```

Comment:
A uring is not thread safe. Do we expect the caller not to make ReadAsync() call in one thread and polling in another? It feels nicer if we can provide an interface that isn't prone to misuse.

Do we consider for a user to pass a async call context that contains an IO Uring?

Reply:
They're supposed to be in the same thread. Do you mean a way to enforce that?The io_handle is context that the ReadAsync() implementation can allocate, and will be passed back to it in Poll(). PosixEnv can store a pointer to the IO uring in it, and compare against thread_local_io_urings_->Get() to detect if the call is being made on a different thread..